packages/ui-components/src/components/DateTimePicker/DateTimePicker.component.tsx

```diff
@@ -0,0 +1,651 @@
+  /** Pass custom classNames. These will be appended to the input element of the DateTimePicker. */
+  className?: string
+  /** A custom string to separate individual dates in `multiple` mode. */
+  conjunction?: string
+  /** A string of characters to customize how a date will be formatted in the input field. Available options: https://flatpickr.js.org/formatting/ */
+  dateFormat?: string
+  /** Sets the default date of the DateTimePicker. Same as `value`, only here for compatibility with the original Flatpickr library. If both `value` and `defaultDate` are being passed, `value` will win. Date Objects, timestamps, ISO date strings, chronological date strings `YYYY-MM-DD HH:MM` (must be compatible to current `dateFormat`), and the shortcut `today` are all accepted. */
+  defaultDate?: DateOption | DateOption[] | null
+  /** The initial value of the hour input element. Only effective if time is enabled. Note this will only set the hour input element to the value specified. Setting this options will not set a selected value on the DateTimePicker. */
+  defaultHour?: number
+  /** The initial value of the minute input element. Only effective if time is enabled. Note this will only set the minute input element to the value specified. Setting this options will not set a selected value on the DateTimePicker. */
+  defaultMinute?: number
+  /** Same as value, defaultDate */
+  defaultValue?: DateOption | DateOption[]
+  /** Pass an array of dates, date strings, date ranges or functions to disable dates. More on disabling dates: https://flatpickr.js.org/examples/#disabling-specific-dates */
+  disable?: DateLimit<DateOption>[]
+  /** Whether the DateTimePicker is disabled */
+  disabled?: boolean
+  /** Whether to show seconds when showing a time picker. */
+  enableSeconds?: boolean
+  /** Whether to show a time picker.  */
+  enableTime?: boolean
+  /** A text to render when the DateTimePicker has an error or could not be validated. */
+  errortext?: React.ReactNode
+  /** A helptext to render to explain meaning and significance of the DateTimePicker. */
+  helptext?: React.ReactNode
+  /** The step for the hour input. Only has an effect when a time picker is enabled via `enableTime`. */
+  hourIncrement?: number
+  /** The id of the DateTimePicker input element. If none is passed, an automatically generated id will be used. */
+  id?: string
+  /** Whether the DateTimePicker selected date was negatively validated.  */
+  invalid?: boolean
+  /** The label of the DateTimePicker input element. */
+  label?: string
+  /** Localization string or object. Can be used to set starting day of the week, e.g. Mondays instead of Sundays. More on localization: https://flatpickr.js.org/localization/ */
+  locale?: LocaleKey | Partial<CustomLocale> | null
+  /** The maximum / latest date a user can select (inclusive). */
+  maxDate?: DateOption | null
+  /** The minimum / earliest date a user can select (inclusive). */
+  minDate?: DateOption | null
```

Comment:
**(Breaking Change)**

`DateOption` is the most accurate type here. But we need to be backwards compatible as we allowed `string | array | object | number` previously. We can remove these in the future (added to improvements).

Check all other types for backward compatibility too, bitte. I see there's a few others. 💪🤪

Reply:
**Update:** I added `DateOption` as an [improvement](https://github.com/cloudoperators/juno/issues/570) for date props in this component. For now, let's just support types that were previously supported and leave it out.

Reply:
Since maxDate and minDate are not being manipulated and are simply passed through to the flatpickr, the DateOption type is appropriate for them.

Reply:
It has been adressed